### ISSUE TITLE
Operation timed out but not retrying

### ISSUE THREAD

**Issue by ion-elgreco** [apache/arrow-rs#5580]
**Describe the bug**
When writing a parquet file with delta-rs using an ObjectStore filehandler, it will sometimes fail when the operation times out, which is fine. The only thing is it should retry which it's not doing.

```
OSError: Generic MicrosoftAzure error: Error after 0 retries in 171.084552191s, max_retries:10, retry_timeout:180s, source:error sending request for url (https://<redacted>.blob.core.windows.net/<redacted>/<redacted>/<redacted>=<redacted>/<redacted>=202110/0-afa3cc5d-d0b7-4258-8baf-e84bca411da2-0.parquet?comp=block&blockid=ICAgICAgICAgICAgICAgICAgIDM%3D): operation timed out
```
**To Reproduce**

**Expected behavior**
Respect the retry amount?

**Comment by tustvold** [comment]
Timeouts are currently only retried automatically on idempotent requests, i.e. reads

**Comment by ion-elgreco** [comment]
@tustvold could be useful to propagate that info in the error msg, that retry_count is ignored because it's a PUT request

**Comment by tustvold** [comment]
Closed by #5609
